<a href="https://colab.research.google.com/github/AshmitaRakshit/Class110/blob/main/Class110.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import plotly.figure_factory as ff
import plotly.graph_objects as go
import statistics
import random
import csv
import pandas as pd
import plotly.express as px

df = pd.read_csv("data.csv")
data = df["temp"].to_list()
mean= statistics.mean(data)
print("populationMean ",mean)
print("population standardDeviation ",statistics.stdev(data))
fig=ff.create_distplot([data],["temp"],show_hist= False)
fig.add_trace(go.Scatter(x=[mean, mean], y=[0, 0.17], mode="lines", name="MEAN"))
fig.show()

populationMean  35.05393111079237
population standardDeviation  5.699825337585306


From the above graph,the data is not normally distributed as it is not a bell curve.
Data sampling is a statistical analysis technique used to select, manipulate and analyse a subset of datapoints to identify patterns and trends in the larger datasets being examined.

In [7]:
#Taking 10 random samples from the whole data(ex:counter=10)
def random_set_of_mean(counter):
    dataset = []
    for i in range(0, counter):
        random_index= random.randint(0,len(data))
        value = data[random_index]
        dataset.append(value)
        #dataset is the list containing 10 random samples from the whole data
    mean_10_samples = statistics.mean(dataset)
    std_deviation_10_samples=statistics.stdev(dataset)

    #fig = ff.create_distplot([dataset], ["mean_10_samples- temp"], show_hist=False)
    #fig.add_trace(go.Scatter(x=[mean_10_samples, mean_10_samples],y=[0,1], mode="lines", name="MEAN"))
    #fig.show()
    #print("Mean of 10 random Sample :- ",mean_10_samples)
    #print("std_deviation of 10 random sample:- ",std_deviation_10_samples)
    
    return mean_10_samples
#------------------------------------------------------------------
def show_fig(mean_list_100):
    df = mean_list_100
    sampling_mean = sum(mean_list_100) / len(mean_list_100)
    fig = ff.create_distplot([df], ["Sampling mean -temp"], show_hist=False)
    fig.add_trace(go.Scatter(x=[sampling_mean, sampling_mean],y=[0,1], mode="lines", name="MEAN"))
    fig.show()

def setup():
    mean_list_100 = []
    for i in range(0,100):
        set_of_means= random_set_of_mean(10)
        #set_of_means is the mean of 10 random data points
        #each time the for loop repeats we will get one mean
        #we are appending this mean to the list mean_list_100
        mean_list_100.append(set_of_means)
    show_fig(mean_list_100)
    print("sampling mean :- ", statistics.mean(mean_list_100))
    print("sampling std_deviation:- ", statistics.stdev(mean_list_100))
setup()

sampling mean :-  35.008
sampling std_deviation:-  1.7398293193891348


We observe population mean and standard mean are nearly equal=35.183
PopulationStandardDeviation  5.699825337585306
StandardDeviation of distribution=1.7398293193891348

We learned how to create a sampling mean distribution of a population for different sample sizes and different sampling sizes - 
We observed how sampling mean distribution for large sample sizes are always normal distribution - 
Sampling Mean = Population Mean 
Sampling Error = Population Standard Deviation / sqrt (sampling size)